### PR TITLE
Allow to store scmRepo object to pipeline

### PR DESCRIPTION
**Pull request by catto**
This PR allows to store scmRepo object to a pipeline. 
(screwdriver-cd/screwdriver#194 and screwdriver-cd/screwdriver#160)

### CODE REVIEW

**Comment by catto** [comment]
bumped the version to 13

File: models/pipeline.js [hunk 1]
```diff
@@ -74,7 +77,8 @@ module.exports = {
      * @type {Joi}
      */
     create: Joi.object(mutate(MODEL, [
-        'scmUrl'
+        'scmUrl',
+        'scmRepo'
```
Comment (stjohnjohnson) [inline models/pipeline.js]: It's not required for create.

File: models/pipeline.js [hunk 2]
```diff
@@ -85,7 +89,7 @@ module.exports = {
      * @property keys
      * @type {Array}
      */
-    keys: ['scmUrl'],
+    keys: ['scmRepo.id'],
```
Comment (stjohnjohnson) [inline models/pipeline.js]: Don't change the key yet, that way it won't be backwards incompatible.  We can slowly migrate towards using it.

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "screwdriver-data-schema",
-  "version": "12.1.0",
+  "version": "13.0.0",
```
Comment (stjohnjohnson) [inline package.json]: If we remove the key change, this is not-backwards incompatible and we can set this back to `12.1.0`

File: models/pipeline.js (hunk 1 above)
Comment (catto) [inline models/pipeline.js]: I moved it to optionalFields

File: models/pipeline.js (hunk 2 above)
Comment (catto) [inline models/pipeline.js]: Reverted this and version

**Pull request merged**